tests/integration-tests/tests/ad_integration/test_ad_integration.py

```diff
@@ -195,14 +207,17 @@ def _create_directory_stack(cfn_stacks_factory, request, directory_type, region,
 
     with open(directory_stack_template_path) as directory_stack_template:
         stack_parameters = _get_stack_parameters(directory_type, vpc_stack, request.config.getoption("key_name"))
+        tags = [{"Key": "parallelcluster:integ-tests-ad-stack", "Value": directory_type}]
+        if request.config.getoption("retain_ad_stack"):
+            tags.append({"Key": "DO-NOT-DELETE", "Value": " "})
```

Comment:
I suggest to use the value to specify the reason why we should not delete it. Could also be as simple as "Retained for integration testing". This would be helpful when someone not aware of your mechanism will see retained stack with this new tag